atef/bin/check.py

```diff
@@ -208,46 +83,137 @@ def log_results_rich(
+            result = item.result
+            prepared = item
+        else:
+            # An error that was transformed into a Result with a severity
+            result = item
+            prepared = None
+
+        if result is None:
+            tree.add(
+                f"{severity_to_rich[Severity.internal_error]}[default]: "
+                f"comparison not run"
+            )
+        elif result.severity > Severity.success:
             tree.add(
                 f"{severity_to_rich[result.severity]}[default]: {result.reason}"
             )
+        elif verbose > 0 and prepared is not None:
+            if prepared.comparison is not None:
+                description = prepared.comparison.describe()
+            else:
+                description = "no comparison configured"
+
+            tree.add(
+                f"{severity_to_rich[result.severity]}[default]: "
+                f"{prepared.identifier} {description}"
+            )
+
     console.print(tree)
 
 
+def check_and_log(
+    config: AnyConfiguration,
+    console: rich.console.Console,
+    verbose: int = 0,
+    client: Optional[happi.Client] = None,
+    name_filter: Optional[Sequence[str]] = None,
+):
+    """Check a configuration and log the results."""
+    items = []
+    name_filter = list(name_filter or [])
+    severities = []
+    for prepared in PreparedComparison.from_config(config, client=client):
+        if isinstance(prepared, PreparedComparison):
+            if name_filter:
+                device_name = getattr(prepared.device, "name", None)
+                if device_name is not None:
+                    if device_name not in name_filter:
+                        logger.debug(
+                            "Skipping device check at user's request: %s",
+                            device_name,
+                        )
+                        continue
+                elif prepared.identifier not in name_filter:
+                    logger.debug(
+                        "Skipping identifier at user's request: %s",
+                        prepared.identifier
+                    )
+                    continue
+
+            prepared.result = prepared.compare()
+            if prepared.result is not None:
+                items.append(prepared)
+                severities.append(prepared.result.severity)
+        elif isinstance(prepared, Exception):
+            ex = cast(Exception, prepared)
+            result = Result.from_exception(ex)
+            items.append(result)
+            severities.append(result.severity)
+        else:
+            logger.error(
+                "Internal error: unexpected result from PreparedComparison: %s",
+                type(prepared)
+            )
+
+    if not items:
+        # Nothing to report; all filtered out
+        return
+
+    log_results_rich(
+        console,
+        config=config,
+        severity=get_maximum_severity(severities),
+        items=items,
+        verbose=verbose,
+    )
```

Comment:
A lot of the stuff in this section above I'll need to play around with manually to get a feel for/opinions on it. Should merge without considering that and follow-up later if needed.